google/cloud/aiplatform/tensorboard/uploader.py

```diff
@@ -678,25 +623,82 @@ def dispatch_requests(
         self._request_sender.flush()
 
 
-class _TimeSeriesResourceManager(object):
+class _OnePlatformResourceManager(object):
     """Helper class managing Time Series resources."""
 
-    def __init__(self, run_resource_id: str, api: TensorboardServiceClient):
-        """Constructor for _TimeSeriesResourceManager.
+    def __init__(self, experiment_resource_name: str, api: TensorboardServiceClient):
+        """Constructor for _OnePlatformResourceManager.
 
         Args:
-          run_resource_id: The resource id for the run with the following format
-            projects/{project}/locations/{location}/tensorboards/{tensorboard}/experiments/{experiment}/runs/{run}
+          experiment_resource_name: The resource id for the run with the following format
+            projects/{project}/locations/{location}/tensorboards/{tensorboard}/experiments/{experiment}
           api: TensorboardServiceStub
         """
-        self._run_resource_id = run_resource_id
+        self._experiment_resource_name = experiment_resource_name
         self._api = api
-        self._tag_to_time_series_proto: Dict[
-            str, tensorboard_time_series.TensorboardTimeSeries
-        ] = {}
+        self._run_name_to_run_resource_name: Dict[str, str] = {}
+        self._run_tag_name_to_time_series_name: Dict[(str, str), str] = {}
+
+    def get_run_resource_name(self, run_name: str):
```

Comment:
better to rename it as create_or_get_run_resource_name ?

Reply:
It doesn't actually "create" a resource name though, but I see what you're saying.

Gonna leave the name as is, but will add doc string to describe what it does.